tapiriik/services/service_record.py

```diff
@@ -42,8 +42,23 @@ def HasExtendedAuthorizationDetails(self, persisted_only=False):
 
     def HasAuthSyncError(self):
         if hasattr(self, "SyncErrors"):
-            sync_errors_with_user_exception = [se for se in self.SyncErrors if "UserException" in se]
-            return next((se for se in sync_errors_with_user_exception if se["UserException"].get("InterventionRequired") and se.get("Block") and se["UserException"].get("Type") == "auth"), False) is not False
+            if type(self.SyncErrors) == list:
+                sync_errors_with_user_exception = [
+                    se for se in self.SyncErrors if 
+                        type(se) == dict and 
+                        "UserException" in se and 
```

Comment:
self.SyncErrors is an array of object

if se is a dict :

```
        {
            "Step": 'upload',
            "Message": 'Could not upload activity 400',
            "Block": True,
            'UserException': {
                "Type": UserExceptionType.Authorization, 
                "InterventionRequired": True
            },
            "Scope": 'service',
            "TriggerExhaustive": False,
            "Timestamp": datetime(2022, 6, 16, 9, 56, 0, 702000)
        },
```

"UserException" in se

```
            'UserException': {
                "Type": UserExceptionType.Authorization, 
                "InterventionRequired": True
            },
```

in keyword works for dict ?